### PR TITLE
[FIX] im_livechat: properly guard inside debounced scroll

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2960982

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js [hunk 1]
```diff
@@ -275,6 +275,9 @@ const PublicLivechatWindow = Widget.extend({
      * @private
      */
     _onScroll() {
+        if (!this.messaging.publicLivechatGlobal.livechatButtonView.chatWindow) {
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:278]: I'm not sure exactly what can exist or not in these new fields, but shouldn't it also guard for other parts of the relation?

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:278]: No need to guard other fields: `livechatButtonView` always exist in this flow, `publicLivechatView` is coupled with `chatWindow`, and `PublicLivechatView/widget` is coupled with the model `PublicLivechatView` itself.

Same remark with `publicLivechat` & `PublicLivechat/legacyPublicLivechat`

Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:278]: But if the scroll is debounced we are potentially in no specific flow? For all we know even messaging itself could be deleted.

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:278]: Indeed, will do proper guard

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**